pyflow/host.py

```diff
@@ -1014,35 +1014,70 @@ def script_submit_arguments(self, submit_arguments):
         """
         Accepted submit arguments:
         """
+
+        TROIKA_0_2_2 = False
+
         resources = {
-            "queue": "--qos=",
-            "job_name": "--job-name=",
-            "tasks": "--ntasks=",
-            "nodes": "--nodes=",
-            "threads_per_task": "--cpus-per-task=",
-            "tasks_per_node": "--ntasks-per-node=",
-            "hyperthreads": "--threads-per-core=",
-            "memory_per_task": "--mem-per-cpu=",
-            "accounting": "--account=",
-            "working_dir": "--chdir=",
-            "time": "--time=",
-            "output": "--output=",
-            "error": "--error=",
-            "priority": "--priority=",
-            "tmpdir": "--gres=ssdtmp:",
-            "sthost": "--export=STHOST=",
+            "queue": "queue",
+            "job_name": "job_name",
+            "tasks": "total_tasks",
+            "nodes": "total_nodes",
+            "threads_per_task": "cpus_per_task",
+            "tasks_per_node": "tasks_per_node",
+            "hyperthreads": "threads_per_core",
+            "memory_per_task": "memory_per_cpu",
+            "accounting": "billing_account",
+            "working_dir": "working_dir",
+            "time": "time",
+            "output": "output",
+            "error": "error",
+            "priority": "priority",
+            "tmpdir": "tmpdir_size",
+            "enable_hyperthreading": "enable_hyperthreading",
+            "distribution": "distribution",  # XXX: troika >=0.2.2
+            "reservation": "reservation",  # XXX: troika >=0.2.2
+        }
+
+        slurm_resources = {
+            "tmpdir": "--gres=ssdtmp:",  # XXX: ECMWF-specific
```

Comment:
Shouldn't this be in the troika config? We would then rely here on the troika tmdir_size definition instead?

Reply:
It should. I left it here for backwards compatibility, but I would very much prefer using the Troika directive with an overload in the local config.